File: GUI_Testing_Automation_CL/Model/Path/ElementPath.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GUI_Testing_Automation
{
    public class ElementPath : IElementPath
    {
        string path;
        public ElementPath(string path)
        {
            this.path = path;
        }
        public const string SLASH = "/";

        public ElementPath AppendWith(string strAppend)
        {
            return new ElementPath(SLASH + strAppend);
        }

        public string Path
        {
            get
            {
                return this.path;
            }
            set
            {
                this.path = value;
            }
        }
    }
}

File: JavaTestMobile/src/com/fit/model/path/ElementPath.java
package com.fit.model.path;

public class ElementPath implements IElementPath {
    protected String path;
    public ElementPath(String path)
    {
        this.path = path;
    }
    public final String CONNECT = ".";

    public ElementPath AppendWith(String strAppend)
    {
        return new ElementPath(CONNECT + strAppend);
    }

    @Override
    public String getPath() {
        return path;
    }

    @Override
    public void setPath(String path) {
        this.path = path;
    }
}
